doc/pull_request_template.md

```diff
@@ -0,0 +1,44 @@
+# Pull Request Template
+
+## Description
+
+Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
+
+Jira link: 
+
+## Type of change
+
+Please delete options that are not relevant.
+
+- [ ] Bug fix (non-breaking change which fixes an issue)
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
+- [ ] This change requires a documentation update
+
+## How Has This Been Tested?
+
+Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
+
+- [ ] Passes test case
+- [ ] Other (please explain)
+
+
+
+**Test Configuration**:
+* Firmware version:
+* Hardware:
+* Toolchain:
+* SDK:
+
+## Checklist:
+
+- [ ] My code follows the style guidelines of this project
```

Comment:
Consider replacing with - I have run the precommit and/or linter

Reply:
Doesn't precommit run automatically when doing `git commit`? Is it just for me?

Reply:
I do not think mine does no. Do you have any extra packages or config installed to do that?

Reply:
hmm I wonder if this is one of those pc/mac things 
![image](https://user-images.githubusercontent.com/30075973/215537303-75323121-5c3a-4990-b507-9fc095017f13.png)

Reply:
Yup that was it -- I guess I only installed api and webapp packages haha

Reply:
So @SayakaOno  would it make sense to change the recommentation to 

Consider replacing with - "The precommit and linting ran successfully"

Reply:
Sounds good! (I was just thinking we could remove it if precommit would run for everyone. Sorry, I'm too lazy...)